src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,348 @@
+
+                // Moving inner imports out of the namespace declaration can lead to a break in semantics.
+                // For example:
+                //
+                //  namespace A.B.C
+                //  {
+                //    using D.E.F;
+                //  }
+                //
+                //  The using of D.E.F is looked up iwith in the context of A.B.C first. If it's moved outside,
+                //  it may fail to resolve.
+
+                return compilationUnit.Update(
+                    compilationUnit.Externs.AddRange(namespaceDeclaration.Externs), 
+                    compilationUnit.Usings.AddRange(namespaceDeclaration.Usings), 
+                    compilationUnit.AttributeLists, 
+                    members,
+                    eofToken);
+            }
+
+            // Change namespace name
+            return compilationUnit.ReplaceNode(namespaceDeclaration, 
+                namespaceDeclaration.WithName(
+                    CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                    .WithTriviaFrom(namespaceDeclaration.Name)));
+        }
+
+        private static string GetAliasQualifierOpt(SyntaxNode name)
+        {
+            while (true)
+            {
+                switch (name.Kind())
+                {
+                    case SyntaxKind.QualifiedName:
+                        name = ((QualifiedNameSyntax)name).Left;
+                        continue;
+                    case SyntaxKind.AliasQualifiedName:
+                        return ((AliasQualifiedNameSyntax)name).Alias.Identifier.ValueText;
+                }
+
+                return null;
+            }
+        }
+
+        private NameSyntax CreateNameSyntax(ImmutableArray<string> namespaceParts, string aliasQualifier, int index)
+        {
+            var part = namespaceParts[index].EscapeIdentifier();
+            Debug.Assert(part.Length > 0);
+
+            var namePiece = SyntaxFactory.IdentifierName(part);
+
+            if (index == 0)
+            {
+                return aliasQualifier == null ? (NameSyntax)namePiece : SyntaxFactory.AliasQualifiedName(aliasQualifier, namePiece);
+            }
+            else
+            {
+                return SyntaxFactory.QualifiedName(CreateNameSyntax(namespaceParts, aliasQualifier, index - 1), namePiece);
+            }
+        }
+
+        /// <summary>
+        /// return trivia attached to namespace declaration. 
+        /// Leading trivia of the node and trivia around opening brace, as well as
+        /// trivia around closing brace are concatenated together respectively.
+        /// </summary>
+        private static (ImmutableArray<SyntaxTrivia> openingTrivia, ImmutableArray<SyntaxTrivia> closingTrivia) 
+            GetOpeningAndClosingTriviaOfNamespaceDeclaration(NamespaceDeclarationSyntax namespaceDeclaration)
+        {
+            var openingBuilder = ArrayBuilder<SyntaxTrivia>.GetInstance();
```

Comment:
so, we don't care comment between tokens? namepsace /**/ I /**/ . /**/ B ...

Reply:
Nope, trivia like that will be lost